yoti-sdk-api/src/main/java/com/yoti/api/client/spi/remote/call/identity/IdentityService.java

```diff
@@ -0,0 +1,98 @@
+package com.yoti.api.client.spi.remote.call.identity;
+
+import static com.yoti.api.client.spi.remote.call.HttpMethod.HTTP_POST;
+import static com.yoti.api.client.spi.remote.call.YotiConstants.AUTH_ID_HEADER;
+import static com.yoti.api.client.spi.remote.call.YotiConstants.CONTENT_TYPE;
+import static com.yoti.api.client.spi.remote.call.YotiConstants.CONTENT_TYPE_JSON;
+import static com.yoti.api.client.spi.remote.call.YotiConstants.DEFAULT_IDENTITY_URL;
+import static com.yoti.api.client.spi.remote.call.YotiConstants.PROPERTY_YOTI_API_URL;
+import static com.yoti.api.client.spi.remote.util.Validation.notNull;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.security.GeneralSecurityException;
+import java.security.KeyPair;
+
+import com.yoti.api.client.identity.SessionCreation;
+import com.yoti.api.client.identity.SessionResult;
+import com.yoti.api.client.spi.remote.call.ResourceException;
+import com.yoti.api.client.spi.remote.call.SignedRequest;
+import com.yoti.api.client.spi.remote.call.SignedRequestBuilderFactory;
+import com.yoti.api.client.spi.remote.call.factory.UnsignedPathFactory;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class IdentityService {
+
+    public static IdentityService newInstance() {
+        return new IdentityService(
+                new ObjectMapper(),
+                new UnsignedPathFactory(),
+                new SignedRequestBuilderFactory()
+        );
+    }
+
+    static final Logger LOG = LoggerFactory.getLogger(IdentityService.class);
+
+    private final ObjectMapper mapper;
+    private final UnsignedPathFactory pathFactory;
+    private final SignedRequestBuilderFactory requestBuilderFactory;
+
+    private final String apiUrl;
+
+    public IdentityService(
+            ObjectMapper mapper,
+            UnsignedPathFactory pathFactory,
+            SignedRequestBuilderFactory requestBuilderFactory) {
+        this.mapper = mapper;
+        this.pathFactory = pathFactory;
+        this.requestBuilderFactory = requestBuilderFactory;
+
+        this.apiUrl = System.getProperty(PROPERTY_YOTI_API_URL, DEFAULT_IDENTITY_URL);
+    }
+
+    public SessionResult createSession(String appId, KeyPair keyPair, SessionCreation sessionCreation)
+            throws IdentityException {
+        notNull(appId, "Application id");
+        notNull(keyPair, "Application key Pair");
+        notNull(sessionCreation, "Session creation");
+
+        LOG.info("Requesting Session Creation");
```

Comment:
This log message is not good... it has no dynamic info, which means out of 1000 requests _all_ of it's instances will be identical, and on top of that it's logged at `info` (and saying a request has been sent is at _most_ a `debug` if not `trace` level message).

Imagine being an integrator and ask yourself if you'd like to see this log message being logged over and over... how useful would that be? Try to log only messages that you could see yourself as an integrator being interested in seeing.

And avoid logging messages that cannot be distinguished between executions (i.e. messages with no runtime-specific information)